### PR TITLE
Unit 3.3 webpack

### LINKED ISSUE #14: 3.3: Webpack
## Main Goals
1. Configure Webpack with Polymer to build out an app.
2. Organize project so that all source code is separate from project metadata and build code.
3. Refactor npm scripts to use Webpack.
### When Accomplished
- You've added new dev dependencies:
  - [x] `webpack`
  - [x] `webpack-cli`
  - [x] `webpack-dev-server`
  - [x] `@babel/core`
  - [x] `babel-loader`
  - [x] `polymer-webpack-loader`
  - [x] `file-loader`
  - [x] `css-loader`
  - [x] `html-webpack-plugin`
  - [x] `copy-webpack-plugin`
- You've added the Banno-scoped Polymer and design components as normal dependencies:
  - [x] `@banno/polymer`
  - [x] `@banno/jha-design-components`
- [x] `webpack.config.js` file exists in the root of the project
- [x] You've added the `jha-design-components-loader` tool in the `tools` directory
- [x] `webpack` command returns no errors
- [x] The `start` script now runs `webpack-dev-server`
- [x] A new script has been added to `package.json` called `build` which simply runs `webpack`
- [x] You've copied over all content from the `webpack-boilerplate` directory in the project resources to your project's `src` directory
- [x] The new file structure of your project matches the file structure outlined [here](https://github.com/Banno/ux-intern-onboarding/blob/master/resources/specs/unit-3.3-project-structure.md).
## Research
1. [Webpack: The Basics -- Hacker Noon](https://hackernoon.com/webpack-the-basics-2712a7ad640b)

## Notes
- Webpack can be installed globally, but it's better to install on a per-project basis.
- `webpack.config.js` is already written so it just needs to be placed in the root of the project.
  - The configuration we use requires a few other files to be added. Make sure to copy over everything from the `webpack-boilerplate` directory--our specific configuration relies on an 'entry' JavaScript file and a 'template' HTML file as well as the Polymer components that will start out the Polymer implementation of your project.
## Resources
- [Webpack boilerplate for this project](https://github.com/Banno/ux-intern-onboarding/tree/master/resources/webpack-boilerplate)
  - Make sure to read through `webpack.config.js`--there are plenty of comments that will help inform your understanding of what Webpack does with your files.
- [Webpack Book](https://survivejs.com/webpack/introduction/)

### PR DESCRIPTION
**Pull request by frankiebarrios**
## What It Does
This pull request is set to complete Unit 3.3 of the ux-onboarding-project.

This work is taking place to complete unit 3.3 of the ux-onboarding-project.

**Implementation:**
- Added webpack.
- Updated scripts to run app.
- Added dependencies required to run with webpack.

[closes #14  ]

## How To Test
- Install dependencies using instruction in README.
- Start server using yarn start or build using yarn build.

## Notes/Caveats

None
## Checklist

Under penalty of public shaming, I avow that I:
**Many items in list N/A** 
- [X] Reviewed the diff to look for typos, whitespace errors, and console.log() statements
- [X] Verified that there are no compiler or linter errors or warnings
- [X] Verified the build in production(optimized) mode
- [X] Updated the docs, if necessary
- [X] Included the appropriate labels
- [X] Referenced applicable issues (i.e. Closes #XXXX, Fixes #XXXX)

Browsers tested:
**Many items in list N/A** 
- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] Internet Explorer

## Dependencies
1. Install Node and npm
2. Install Yarn
3. Install Ux-Lint

### CODE REVIEW

**Review by kddrake — CHANGES_REQUESTED** [review]
You'll want to keep you `dist/` folder out of your repository... so delete the files and commit, then add `dist/` to your `.gitignore`

let's also get rid of the `starter-component` folder. I know the directions say to include this but I don't feel that's true at this point.

File: package.json [hunk 1]
```diff
@@ -19,12 +19,21 @@
...
   "devDependencies": {
+    "@babel/core": "^7.4.5",
+    "babel-loader": "^8.0.6",
+    "copy-webpack-plugin": "^5.0.3",
+    "css-loader": "^3.0.0",
     "eslint": "^6.0.1",
+    "eslint-plugin-html": "^5.0.5",
+    "file-loader": "^4.0.0",
+    "html-webpack-plugin": "^3.2.0",
     "http-server": "^0.11.1",
```
Comment (Jropp) [inline package.json]: Lets delete this now that you are using WP dev server

**Review by Jropp — CHANGES_REQUESTED** [review]
Looks great!

File: package.json (hunk 1 above)
Comment (frankiebarrios) [reply package.json]: Are we talking lines 30-32 or just the http-server?

Comment (Jropp) [reply package.json]: @frankielb just the http-server

Comment (frankiebarrios) [reply package.json]: Done

**Review by Jropp — APPROVED** [review]

**Pull request merged**